contracts/core/external/lib/ZeroExOrderDataHandler.sol

```diff
@@ -0,0 +1,194 @@
+        bytes memory order = _orderData.slice(orderStartAddress, orderStartAddress.add(_orderLength));
+        return order;
+    }
+
+    function parseZeroExOrder(bytes _zeroExOrder, uint _makerAssetDataLength, uint _takerAssetDataLength)
+        internal
+        pure
+        returns (Order memory)
+    {
+        
+        Order memory order;
+        uint256 orderDataAddr = _zeroExOrder.contentAddress();
+
+
+        // | Data     |        | 12 * 32 | order:                                      |
+        // |          | 0x000  |         |   1.  senderAddress                         |
+        // |          | 0x020  |         |   2.  makerAddress                          |
+        // |          | 0x040  |         |   3.  takerAddress                          |
+        // |          | 0x060  |         |   4.  feeRecipientAddress                   |
+        // |          | 0x080  |         |   5.  makerAssetAmount                      |
+        // |          | 0x0A0  |         |   6.  takerAssetAmount                      |
+        // |          | 0x0C0  |         |   7.  makerFeeAmount                        |
+        // |          | 0x0E0  |         |   8.  takerFeeAmount                        |
+        // |          | 0x100  |         |   9.  expirationTimeSeconds                 |
+        // |          | 0x120  |         |   10. salt                                  |
+        // |          | 0x140  |         |   11. Offset to makerAssetData (*)          |
+        // |          | 0x160  |         |   12. Offset to takerAssetData (*)          |
+        // |          | 0x180  | 32      | makerAssetData Length                       | - NOT IMPLEMENTED
+        // |          | 0x1A0  | **      | makerAssetData Contents                     | - NOT IMPLEMENTED`
+        // |          | 0x1C0  | 32      | takerAssetData Length                       | - NOT IMPLEMENTED
+        // |          | 0x1E0  | **      | takerAssetData Contents                     | - NOT IMPLEMENTED
+        assembly {
+            mstore(order,           mload(orderDataAddr))  // maker
+            mstore(add(order, 32),  mload(add(orderDataAddr, 32)))  // taker
+            mstore(add(order, 64),  mload(add(orderDataAddr, 64)))  // feeRecipient
+            mstore(add(order, 96),  mload(add(orderDataAddr, 96)))  // senderAddress
+            mstore(add(order, 128),  mload(add(orderDataAddr, 128))) // makerAssetAmount
+            mstore(add(order, 160), mload(add(orderDataAddr, 160))) // takerAssetAmount
+            mstore(add(order, 192), mload(add(orderDataAddr, 192))) // makerFee
+            mstore(add(order, 224), mload(add(orderDataAddr, 224))) // takerFee
+            mstore(add(order, 256), mload(add(orderDataAddr, 256))) // expirationUnixTimestampSec
+            mstore(add(order, 288), mload(add(orderDataAddr, 288))) // salt
+        }
+
+        order.makerAssetData = _zeroExOrder.slice(320, _makerAssetDataLength.add(320));
```

Comment:
What's 320? can you make it into a constant?

Reply:
I'll add / update a table